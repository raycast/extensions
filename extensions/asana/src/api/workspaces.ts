import { request } from "./request";

export type Workspace = {
  gid: string;
  name: string;
};

export async function getWorkspaces() {
  const { data } = await request<{ data: Workspace[] }>("/workspaces");

  return data.data;
}
