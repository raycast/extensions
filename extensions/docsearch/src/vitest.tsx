import { SearchDocumentation } from "./components";

export default function Command(props: { arguments: { search?: string } }) {
  return <SearchDocumentation docsName="Vitest" quickSearch={props.arguments?.search} />;
}
