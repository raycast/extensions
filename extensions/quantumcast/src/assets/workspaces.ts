import { cloudflowBaseUrl, cloudflowPluginSuiteBaseUrl } from "./preferences";
import { Workspace } from "../types"

/**
 * Workspaces
 */

const workspaces: Workspace[] = [
	{
		name: "Home",
		url: `${cloudflowBaseUrl}/portal.cgi?home=1`
	},
	{
		name: "Notifications",
		url: `${cloudflowBaseUrl}/portal.cgi?notifications=1`
	},
	{
		name: "Kiosk",
		url: `${cloudflowBaseUrl}/portal.cgi?hub&topbar=true`
	},
	{
		name: "Assets",
		url: `${cloudflowBaseUrl}/portal.cgi?asset=X`
	},
	{
		name: "Flows",
		url: `${cloudflowBaseUrl}/portal.cgi?quantum`
	},
	{
		name: "Approval Templates",
		url: `${cloudflowBaseUrl}/portal.cgi?chains=1#emailtemplates`
	},
	{
		name: "All Approvals",
		url: `${cloudflowBaseUrl}/portal.cgi?chains=1#all_approvals`
	},
	{
		name: "My Approvals",
		url: `${cloudflowBaseUrl}/portal.cgi?chains=1#my_approvals`
	},
	{
		name: "Team Approvals",
		url: `${cloudflowBaseUrl}/portal.cgi?chains=1#team_approvals`
	},
	{
		name: "All Tasks",
		url: `${cloudflowBaseUrl}/?task_list=allTasks`
	},
	{
		name: "My Task",
		url: `${cloudflowBaseUrl}/?task_list=myTasks`
	},
	{
		name: "All Callendars",
		url: `${cloudflowBaseUrl}/?calendar=allCalendars`
	},
	{
		name: "Jobs - Configuration",
		url: `${cloudflowBaseUrl}/?job=configuration`
	},
	{
		name: "Jobs - Form Manager",
		url: `${cloudflowBaseUrl}/?form=editor`
	},
	{
		name: "Jobs - Teams",
		url: `${cloudflowBaseUrl}/?team=teams`
	},
	{
		name: "Jobs - Resources",
		url: `${cloudflowBaseUrl}/?job=resource.configuration`
	},
	{
		name: "RIP - Jobs",
		url: `${cloudflowBaseUrl}/standalone_rip_nebula.html#jobs`
	},
	{
		name: "RIP - Presets",
		url: `${cloudflowBaseUrl}/standalone_rip_nebula.html#presets`
	},
	{
		name: "RIP - Curves",
		url: `${cloudflowBaseUrl}/standalone_rip_nebula.html#curves`
	},
	{
		name: "RIP - Profiles",
		url: `${cloudflowBaseUrl}/standalone_rip_nebula.html#colorprofiles`
	},
	{
		name: "MARS",
		url: `${cloudflowBaseUrl}/appsnebula/index.html`
	},
	{
		name: "Manual",
		url: `${cloudflowBaseUrl}/manual/manual.html`
	},
	{
		name: "API - Javascript",
		url: `${cloudflowBaseUrl}/portal.cgi?api=getJavaScriptDocumentation`
	},
	{
		name: "API - Web Service",
		url: `${cloudflowBaseUrl}/portal.cgi?api=getWebServiceDocumentation`
	},
	{
		name: "Inspect Collection",
		url: `${cloudflowBaseUrl}/mongoCloud/mongoCloud_nebula.html`
	},
	{
		name: "Standalone VDP",
		url: `${cloudflowBaseUrl}/portal.cgi/__CloudflowApps/VDP/jobStation.html`
	},
	{
		name: "Tectonics - Queues",
		url: `${cloudflowBaseUrl}/portal.cgi/__CloudflowApps/tectonics/application/queues-main.html`
	},
	{
		name: "Patchplanner",
		url: `${cloudflowBaseUrl}/portal.cgi?patchplanner=1`
	},
	{
		name: "Dashboard - Main",
		url: `${cloudflowBaseUrl}/portal.cgi?portal=dashboard`
	},
	{
		name: "Dashboard - Proofscope Live Renderers",
		url: `${cloudflowBaseUrl}/portal.cgi?proofscope=live_renderers`
	},
	{
		name: "Dashboard - Logs",
		url: `${cloudflowBaseUrl}/portal.cgi?logging=1`
	},
	{
		name: "Dashboard - Workables",
		url: `${cloudflowBaseUrl}/portal.cgi?config=1#workables`
	},
	{
		name: "Main Settings",
		url: `${cloudflowBaseUrl}/portal.cgi?config=1#settings`
	},
	{
		name: "Contacts",
		url: `${cloudflowBaseUrl}/portal.cgi?manageusers=1`
	},
	{
		name: "Users",
		url: `${cloudflowBaseUrl}/portal.cgi?manageusers=1&usersonly=true`
	},
	{
		name: "Scopes",
		url: `${cloudflowBaseUrl}/portal.cgi?config=1#scopes`
	},
	{
		name: "Share - Sites",
		url: `${cloudflowBaseUrl}/portal.cgi?config=1#share_sites`
	},
	{
		name: "Share - Sync Specifications",
		url: `${cloudflowBaseUrl}/portal.cgi?share=1`
	},
	{
		name: "Share - Activities",
		url: `${cloudflowBaseUrl}/portal.cgi?config=1#share_activity`
	},
	{
		name: "DataLink",
		url: `${cloudflowBaseUrl}/portal.cgi/__CloudflowApps/datalink/datalink.html`
	},
	{
		name: "Filestores",
		url: `${cloudflowBaseUrl}/portal.cgi?config=1#filestores`
	},
	{
		name: "Folder Mappings",
		url: `${cloudflowBaseUrl}/portal.cgi?config=1#foldermappingtable`
	},
	{
		name: "Work Servers",
		url: `${cloudflowBaseUrl}/portal.cgi?config=1#workers`
	},
	{
		name: "Resources",
		url: `${cloudflowBaseUrl}/portal.cgi?config=1#resources`
	},
	{
		name: "Paper Definitions",
		url: `${cloudflowBaseUrl}/portal.cgi?config=1#outputdevices`
	},
	{
		name: "System Tools",
		url: `${cloudflowBaseUrl}/portal.cgi?config=1#maintenance`
	},
	{
		name: "License",
		url: `${cloudflowBaseUrl}/portal.cgi?config=1#license`
	},
	{
		name: "User Preferences",
		url: `${cloudflowBaseUrl}/?user_preferences=1`
	},
	{
		name: "Plug-In Suite Preferences",
		url: `${cloudflowPluginSuiteBaseUrl}/frameCGI/Resource/Web/Frame/preferences_nebula.html`
	},
] 

export default workspaces
