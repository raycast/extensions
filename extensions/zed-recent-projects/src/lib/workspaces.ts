//
// Zed Workspace types, returned from query
//
export type ZedWorkspaceType = "local" | "remote";

export interface ZedBaseWorkspace {
  id: number;
  timestamp: number;
  type: ZedWorkspaceType;
  paths: string;
}

export interface ZedLocalWorkspace extends ZedBaseWorkspace {
  type: "local";
}

export interface ZedRemoteWorkspace extends ZedBaseWorkspace {
  type: "remote";
  host: string;
  user: string | null;
  port: number | null;
}

export type ZedWorkspace = ZedLocalWorkspace | ZedRemoteWorkspace;

//
// Unified types for extension
//

export interface Workspace {
  id: number;
  lastOpened: number;
  type: ZedWorkspaceType;
  path: string;
  uri: string;
  host?: string;
}

export function parseZedWorkspace(zedWorkspace: ZedWorkspace): Workspace | null {
  if (!zedWorkspace.paths) {
    return null;
  }

  const paths = zedWorkspace.paths
    .split("\n")
    .map((p) => p.trim())
    .filter((p) => p);

  if (paths.length !== 1) {
    return null;
  }

  const path = paths[0];

  if (zedWorkspace.type === "local") {
    const processedPath = path.replace(/\/+$/, "");
    return {
      id: zedWorkspace.id,
      lastOpened: zedWorkspace.timestamp,
      type: zedWorkspace.type,
      uri: "file://" + processedPath,
      path: processedPath,
    };
  }

  if (zedWorkspace.type === "remote") {
    const processedPath = path.replace(/^\/+/, "").replace(/\/+$/, "");
    const uri = `ssh://${zedWorkspace.user ? zedWorkspace.user + "@" : ""}${zedWorkspace.host}${
      zedWorkspace.port ? ":" + zedWorkspace.port : ""
    }/${processedPath}`;

    return {
      id: zedWorkspace.id,
      lastOpened: zedWorkspace.timestamp,
      type: zedWorkspace.type,
      uri,
      path: processedPath,
      host: zedWorkspace.host,
    };
  }

  return null;
}
