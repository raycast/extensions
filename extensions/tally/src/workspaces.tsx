import { Action, ActionPanel, Form, Icon, List, showToast, Toast, useNavigation } from "@raycast/api";
import { API_HEADERS, API_URL, useTallyPaginated } from "./tally";
import { Workspace } from "./interfaces";
import OpenInTally from "./open-in-tally";
import { FormValidation, useForm } from "@raycast/utils";

export default function Workspaces() {
  const { isLoading, data: workspaces, revalidate } = useTallyPaginated<Workspace>("workspaces");
  return (
    <List isLoading={isLoading}>
      {workspaces.map((workspace) => (
        <List.Item
          key={workspace.id}
          icon={Icon.Folder}
          title={workspace.name || "My workspace"}
          accessories={[
            { icon: Icon.TwoPeople, text: `${workspace.members.length} members` },
            { icon: Icon.Envelope, text: `${workspace.invites.length} invites` },
          ]}
          actions={
            <ActionPanel>
              <OpenInTally route={`workspaces/${workspace.id}`} />
              <Action.Push
                icon={Icon.Pencil}
                title="Update Workspace"
                target={<RenameWorkspace workspace={workspace} onRename={revalidate} />}
              />
            </ActionPanel>
          }
        />
      ))}
    </List>
  );
}

function RenameWorkspace({ workspace, onRename }: { workspace: Workspace; onRename: () => void }) {
  const { pop } = useNavigation();
  const { handleSubmit, itemProps } = useForm<{ name: string }>({
    async onSubmit(values) {
      const toast = await showToast(Toast.Style.Animated, "Renaming");
      try {
        const response = await fetch(API_URL + `workspaces/${workspace.id}`, {
          method: "PATCH",
          headers: API_HEADERS,
          body: JSON.stringify(values),
        });
        if (!response.ok) {
          const err = (await response.json()) as { message: string };
          throw new Error(err.message);
        }
        toast.style = Toast.Style.Success;
        toast.title = "Renamed";
        toast.message = `"${workspace.name}" -> "${values.name}"`;
        onRename();
        pop();
      } catch (error) {
        toast.style = Toast.Style.Failure;
        toast.title = "Could not rename";
        toast.message = `${error}`;
      }
    },
    initialValues: {
      name: workspace.name || "My workspace",
    },
    validation: {
      name: FormValidation.Required,
    },
  });
  return (
    <Form
      actions={
        <ActionPanel>
          <Action.SubmitForm icon={Icon.Check} title="Complete" onSubmit={handleSubmit} />
        </ActionPanel>
      }
    >
      <Form.TextField title="Workspace name" placeholder="My workspace" {...itemProps.name} />
    </Form>
  );
}
