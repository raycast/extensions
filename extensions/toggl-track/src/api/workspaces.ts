import { get } from "@/api/togglClient";

export function getMyWorkspaces() {
  return get<Workspace[]>("/me/workspaces");
}

// https://developers.track.toggl.com/docs/api/workspaces#response-4
export interface Workspace {
  id: number;
  name: string;
  premium: boolean;
  admin: boolean;
  logo_url: string;
  /** Workspace on Premium subscription */
  business_ws: boolean;
  role: "admin" | "projectlead" | "teamlead" | "user";
  only_admins_may_create_projects: boolean;
  only_admins_may_create_tags: boolean;
  organization_id: number;
  projects_billable_by_default: boolean;
  projects_private_by_default: boolean;
}
