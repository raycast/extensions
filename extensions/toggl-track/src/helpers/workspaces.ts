import { Workspace } from "@/api";

export function getWorkspaceById(workspaces: Workspace[], workspaceId: Workspace["id"]) {
  return workspaces.find((workspace) => workspace.id === workspaceId);
}
