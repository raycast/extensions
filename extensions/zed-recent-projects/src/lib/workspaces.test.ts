import { describe, it, expect } from "vitest";
import { parseZedWorkspace } from "./workspaces";

describe("parseZedWorkspace", () => {
  it("parse local workspace with single path", () => {
    expect(
      parseZedWorkspace({
        id: 1,
        type: "local",
        timestamp: 1757750879526,
        paths: `/Users/raycast/Projects/zed-project-1`,
      }),
    ).toMatchSnapshot();
  });

  it("skips local workspace with multiple paths", () => {
    expect(
      parseZedWorkspace({
        id: 1,
        type: "local",
        timestamp: 1757750879526,
        paths: `/Users/raycast/Projects/zed-project-1
/Users/raycast/Projects/zed-project-2`,
      }),
    ).toMatchSnapshot();
  });

  it("parse remote workspace with single path, no user, no port", () => {
    expect(
      parseZedWorkspace({
        id: 1,
        type: "remote",
        timestamp: 1757750879526,
        host: "example.com",
        user: null,
        port: null,
        paths: "/home/user/project",
      }),
    ).toMatchSnapshot();
  });

  it("parse remote workspace with single path, with user, no port", () => {
    expect(
      parseZedWorkspace({
        id: 1,
        type: "remote",
        timestamp: 1757750879526,
        host: "example.com",
        user: "testuser",
        port: null,
        paths: "/home/user/project",
      }),
    ).toMatchSnapshot();
  });

  it("parse remote workspace with single path, no user, with port", () => {
    expect(
      parseZedWorkspace({
        id: 1,
        type: "remote",
        timestamp: 1757750879526,
        host: "example.com",
        user: null,
        port: 22,
        paths: "/home/user/project",
      }),
    ).toMatchSnapshot();
  });

  it("parse remote workspace with single path, with user and port", () => {
    expect(
      parseZedWorkspace({
        id: 1,
        type: "remote",
        timestamp: 1757750879526,
        host: "example.com",
        user: "testuser",
        port: 22,
        paths: "/home/user/project",
      }),
    ).toMatchSnapshot();
  });

  it("skips remote workspace with multiple paths", () => {
    expect(
      parseZedWorkspace({
        id: 1,
        type: "remote",
        timestamp: 1757750879526,
        host: "example.com",
        user: "testuser",
        port: 22,
        paths: "/home/user/project1\n/home/user/project2",
      }),
    ).toMatchSnapshot();
  });

  it("returns null for empty paths", () => {
    expect(
      parseZedWorkspace({
        id: 1,
        type: "local",
        timestamp: 1757750879526,
        paths: "",
      }),
    ).toBe(null);
  });
});
