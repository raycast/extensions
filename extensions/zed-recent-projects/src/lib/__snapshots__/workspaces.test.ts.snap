// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parseZedWorkspace > parse local workspace with single path 1`] = `
{
  "id": 1,
  "lastOpened": 1757750879526,
  "path": "/Users/raycast/Projects/zed-project-1",
  "type": "local",
  "uri": "file:///Users/raycast/Projects/zed-project-1",
}
`;

exports[`parseZedWorkspace > parse remote workspace with single path, no user, no port 1`] = `
{
  "host": "example.com",
  "id": 1,
  "lastOpened": 1757750879526,
  "path": "home/user/project",
  "type": "remote",
  "uri": "ssh://example.com/home/user/project",
}
`;

exports[`parseZedWorkspace > parse remote workspace with single path, no user, with port 1`] = `
{
  "host": "example.com",
  "id": 1,
  "lastOpened": 1757750879526,
  "path": "home/user/project",
  "type": "remote",
  "uri": "ssh://example.com:22/home/user/project",
}
`;

exports[`parseZedWorkspace > parse remote workspace with single path, with user and port 1`] = `
{
  "host": "example.com",
  "id": 1,
  "lastOpened": 1757750879526,
  "path": "home/user/project",
  "type": "remote",
  "uri": "ssh://testuser@example.com:22/home/user/project",
}
`;

exports[`parseZedWorkspace > parse remote workspace with single path, with user, no port 1`] = `
{
  "host": "example.com",
  "id": 1,
  "lastOpened": 1757750879526,
  "path": "home/user/project",
  "type": "remote",
  "uri": "ssh://testuser@example.com/home/user/project",
}
`;

exports[`parseZedWorkspace > skips local workspace with multiple paths 1`] = `null`;

exports[`parseZedWorkspace > skips remote workspace with multiple paths 1`] = `null`;
